### PR TITLE
drivers: gpio: Add support stellaris gpio controller driver

### PR DESCRIPTION
**Pull request by ashokkumar11**

### CODE REVIEW

File: drivers/gpio/gpio_stellaris.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/*
```
Comment (galak) [inline drivers/gpio/gpio_stellaris.h]: Do we need this file, can we just merge it with gpio_stelaris.c?

File: dts/arm/ti/lm3s6965.dtsi [hunk 2]
```diff
@@ -47,6 +47,69 @@
 			status = "disabled";
 			label = "UART_2";
 		};
+
+		gpio0: gpio@40004000 {
+			compatible = "ti,stellaris-gpio";
+			reg = <0x40004000 0xFFF>;
```
Comment (galak) [inline dts/arm/ti/lm3s6965.dtsi]: `reg = <0x40004000 0xFFF>;`

should be:

`<0x40004000 0x1000>;`

Same comment for all other regs

File: drivers/gpio/gpio_stellaris.c [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+	GPIO_DEN_OFFSET = 0x51C,
+	GPIO_IS_OFFSET = 0x404,
+	GPIO_IBE_OFFSET = 0x408,
+	GPIO_IEV_OFFSET = 0x40C,
+	GPIO_IM_OFFSET = 0x410,
+	GPIO_MIS_OFFSET = 0x418,
+	GPIO_ICR_OFFSET = 0x41C,
+};
+
+static u32_t portx_pins[] = {0xFF, 0xFF, 0xFF,
```
Comment (galak) [inline drivers/gpio/gpio_stellaris.c]: It looks like the only reason for port in gpio_stellaris_config is to lookup portx_pins[], so why not just put portx_pins in gpio_stellaris_config, and remove gpio_ports

**Review by galak — CHANGES_REQUESTED** [review]
Some changes requested

File: drivers/gpio/gpio_stellaris.h (hunk 1 above)
Comment (ashokkumar11) [reply drivers/gpio/gpio_stellaris.h]: fixed

File: dts/arm/ti/lm3s6965.dtsi (hunk 2 above)
Comment (ashokkumar11) [reply dts/arm/ti/lm3s6965.dtsi]: fixed

File: drivers/gpio/gpio_stellaris.c (hunk 3 above)
Comment (ashokkumar11) [reply drivers/gpio/gpio_stellaris.c]: moved inside config structure

**Comment by ashokkumar11** [comment]
@galak A gentle reminder

**Review by galak — APPROVED** [review]

**Pull request merged**